examples/14_ampere_tf32_tensorop_gemm/ampere_tf32_tensorop_gemm_cute.cpp

```diff
@@ -42,6 +42,10 @@
 #include "cutlass/util/device_memory.h"
 #include "helper.h"
 
+#if defined(CUTLASS_ENABLE_MKL)
+#  include  <oneapi/mkl.hpp>
```

Comment:
you are building oneMKL interface for Nvidia GPU, you are linking it to the mkl close source for Intel even if it is on Nvida? what is your mechanism to have a correct linking and make sure that the oneMKL does not run on Intel CPU instead of calling cublas behind the scene to go to GPU. If you are always testing it on the IntelCPU there is no need to have  this :  
```c++
  -DENABLE_MKLGPU_BACKEND=${ENABLE_MKLGPU_BACKEND}
    -DENABLE_CUBLAS_BACKEND=${ENABLE_CUBLAS_BACKEND}
    ```

Reply:
> what is your mechanism to have a correct linking and make sure that the oneMKL does not run on Intel CPU

From what I understood about the oneMKL interface, the dispatch is in accordance to the queue. Whether it is a GPU or a CPU queue, and if a GPU queue, then what backend it will select, depending on the Device. 
Since our queue will never be a CPU queue, It will never go to the CPU BLAS side of things.

Reply:
The oneMKL does not have all the backend unless you provide, for example if you dont give a correct pass to the backed there is no GPU or nvidia backend. OneMKL interface is a wrapper over 3 close source actual implementation, meaning oneMKL product, cuBLAs and rocBLAS. if the backend is not available it does not link to the correct one.  the oneMKL close source product is only for intel CPU and GPU

Reply:
> if the backend is not available it does not link to the correct one. the oneMKL close source product is only for intel CPU and GPU, so when you are running on Nvidia GPU you dont need the oneMKL product backend at all, similarly, when running on Intel GPU you dont need the cublas Backend at all

Only one backend is being built. CPU backend is explicitly turned off, and either of the CUBLAS or the MKL_GPU (closed source oneMKL) backend is being built, [here](https://github.com/AD2605/cutlass-fork/blob/e632ea02e69587c1af7fd2ef5aa793df92de16d3/cmake/onemkl.cmake#L42-L50), in accordance to the SYCL target being passed.